dev/test/bulk-writer.ts

```diff
@@ -601,6 +651,46 @@ describe('BulkWriter', () => {
     });
   });
 
+  it('retries retryable batchWrite failures', async () => {
```

Comment:
Is it possible to make these two tests more distinguishable?

Something like:

`retries batchWrite when single operation fails` vs `retries batchWrite when entire RPC fails`.

Furthermore, should we test what happens when we have both non-rertyable and retryable writes?